pkg/coalescer/coalescer_test.go

```diff
@@ -0,0 +1,30 @@
+package coalescer
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestRun(t *testing.T) {
+	sharedVar := 0
+	wg := &sync.WaitGroup{}
+	c := NewCoalescer(func() error {
+		fmt.Printf("Hello\n")
+		time.Sleep(time.Millisecond * 50)
```

Comment:
This test will flake if the scheduler decides to take more than 50ms to run the later `go func()` calls. Let's make this 250ms or so, long enough to not flake but also not be annoying when run in parallel but